### ISSUE TITLE
Learn Git

### ISSUE THREAD

**Issue by webknjaz** [kpi-web-guild/django-girls-blog-MaksMaidaanov#4]
![](https://i.pinimg.com/564x/25/73/14/257314878858202f4cf6b9ca37981cf0.jpg)

Please read&learn the following:

* [x] [ProGit 2](https://git-scm.com/book/en/v2) [🇺🇦](https://git-scm.com/book/uk/v2)
  * [x] 1. Getting Started
  * [x] 2. Git Basics
  * [x] 3. Git Branching
  * [x] 4. Git on the Server
  * [x] 5. Distributed Git
  * [x] 6. GitHub
  * [x] 7. Git Tools
  * [x]  8. Customizing Git
  * [x]  9. Git and other systems
  * [x]  10. Git Internals
* [x] [Try Git Branching](http://learngitbranching.js.org/)
* [x] [Interactive diagram/cheatsheet about git](http://ndpsoftware.com/git-cheatsheet.html)
* [x] [Working on two git branches at once with git worktree](https://andrewlock.net/working-on-two-git-branches-at-once-with-git-worktree/)
* [x] [You're using git wrong](https://web.archive.org/web/20181029051129/https://dpc.pw/blog/2017/08/youre-using-git-wrong/)
* [x] [Learn git concepts, not commands](https://dev.to/unseenwizzard/learn-git-concepts-not-commands-4gjc)
* [x] [The advantages of an email-driven git workflow](https://drewdevault.com/2018/07/02/Email-driven-git.html)
* [x] [Git For Ages 4 And Up](https://youtu.be/1ffBJ4sVUb4)

Extras:
* [x] [A useful template for commit messages](http://codeinthehole.com/tips/a-useful-template-for-commit-messages/)
* [x] [Understanding the GitHub Flow](https://guides.github.com/introduction/flow/)
* [x] [Closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/)
* [x] [email + git = <3](https://git-send-email.io/)

**Comment by MaksMaidaanov** [comment]
I overestimated my abilities a little. I didn’t master a whole book in a day, nothing, tomorrow will be a day and there will be food.

**Comment by MaksMaidaanov** [comment]
Checklist for today:
In theory, I can manage an organization. (only theory and only github)
In theory, I can close the code and even make a branch out of it, gradually commit it and make it follow the main one. On a rat, drain it locally and laugh like an evil genius. (In theory, I'm a rat)
In theory, I understand what a binary search for a bug is, more precisely, instructions on how to apply it. (I feel like a naturalist biologist looking for bugs on branches)
In practice, it is now three o'clock in the morning, and in theory I am a rat.
P.S. There was a joke about a rat

**Comment by MaksMaidaanov** [comment]
Checklist for today:
After reading 8-10 chapters, I realized that sysadmins are not recognized heroes of our time.
I realized that hooks are good, but without them it’s somehow sad and in general I understood what hooks are.
I realized that in addition to botany, I’m starting to understand a little about plumbing and the use of porcelain associated with this craft. (Only a fool would doubt that they are not interconnected)
I realized that the internal architecture of the git was made by worthy people who tolerate other minorities (chapter 9)
It seems that everything was done today, as usual three in the morning and the thought of plumbing and porcelain does not leave me, the feeling that something from Charles Bukovsky.
